lib/spack/spack/solver/asp.py

```diff
@@ -3447,8 +3470,21 @@ def __init__(self, msg):
 
 
 class InternalConcretizerError(spack.error.UnsatisfiableSpecError):
-    """
-    Subclass for new constructor signature for new concretizer
+    """Errors that indicate a bug in Spack."""
+
+    def __init__(self, msg):
+        super(spack.error.UnsatisfiableSpecError, self).__init__(msg)
```

Comment:
I didn't notice this, but it's... weird. I'll follow up in another PR, but we are skipping one class in the MRO because we don't match its conctructor. I wonder why not inheriting from `spack.error.SpecError` then.